FirebaseAuth/Sources/Swift/AuthProvider/OAuthProvider.swift

```diff
@@ -91,23 +91,12 @@ import Foundation
   /// - Parameter accessToken: The access token associated with the Auth credential be created, if
   /// available.
   /// - Returns: An AuthCredential for the specified provider ID, ID token and access token.
-  @available(swift, introduced: 100.0)
-  @objc(credentialWithProviderID:IDToken:accessToken:)
-  public static func credential(withProviderID providerID: NSString,
-                                idToken: NSString,
-                                accessToken: NSString?) -> OAuthCredential {
-    return OAuthCredential(
-      withProviderID: providerID as String,
-      idToken: idToken as String,
-      accessToken: accessToken as String?
-    )
-  }
-
   @available(
-    *,
-    deprecated,
-    message: "Use `credential(providerID: AuthProviderID, idToken: String, accessToken: String) -> OAuthCredential` instead."
+    swift,
+    deprecated: 0.01,
+    message: "Use `credential(providerID: AuthProviderID, idToken: String, accessToken: String? = nil) -> OAuthCredential` instead."
   )
+  @objc(credentialWithProviderID:IDToken:accessToken:)
```

Comment:
After looking at it again, I realized we can avoid some redundancy by instead deprecating the ObjC API for Swift from very early on (wouldn't let me do Swift 0.0). When it's time to remove this API from Swift for good, we can swap the available decorator for `@available(swift, introduced: 100.0)`.

@paulb777 @andrewheard